### PR TITLE
Added C Books

### PR DESCRIPTION
**Pull request by pratyush-patwa**
C books added and issue #4971 Solved

## What does this PR do?
Add Resources

### Why is this valuable (or not)?
Yes the Book by The creater of C language , Dennis M. Ritchie is Added.

### How do we know it's really free?
Its in pdf format.

### For book lists, is it a book? For course lists, is it a course? etc.
Yes It is a Book.

### Checklist:
- [ ] Not a duplicate
- [ ] Included author(s) if appropriate
- [ ] Lists are in alphabetical order
- [ ] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

**Comment by SethiPandi** [comment]
Heads up, if you use the word "fixes" before referring to a issue, it'll automatically close the issue when the PR gets merged.

It looks like Learn C the Hard Way is _not_ free:  
https://www.oreilly.com/library/view/learn-c-the/9780133124385/

This isn't an appropriate source for The C Programming Language, https://inspirit.net.in/ just download and share PDFs without authorization, this is clear from the fact they have books from series like Head First openly available to download.

The book "The Development of the C Language" on Bell Labs seems cool, though!

File: free-programming-books.md [hunk 1]
```diff
@@ -1068,8 +1069,10 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Programming in C](https://www.freetechbooks.com/programming-in-c-t1337.html) - Kishori Mundargi
 * [Structures and C](https://www.smashwords.com/books/view/644937) - Prakash Hegade
 * [The C book](http://publications.gbdirect.co.uk/c_book/) - Mike Banahan, Declan Brady and Mark Doran (PDF, HTML)
+* [The C Programming Language](https://inspirit.net.in/pdfviewer/web/viewer.html?file=..%2F..%2F.%2Fbooks%2Fc%20plus%20plus%2FThe%20C%20Programming%20Language%20-%20Brian%20Kernighan%20%26%20Dennis%20Ritchie.pdf) - Dennis M. Ritchie, Brian W. Kernighan (PDF)
```
Comment (eshellman) [inline free-programming-books.md:1072]: Not free. I asked Dennis about the book; he said he was working on a way to allow authorized free downloads, but AFAIK, it hasn't happened yet. Please remove this.

File: free-programming-books.md [hunk 2]
```diff
@@ -1059,6 +1059,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Everything you need to know about pointers in C - Peter Hosey](http://boredzo.org/pointers/)
 * [Functional C (1997)](https://research.utwente.nl/files/5128727/book.pdf) - Pieter H. Hartel, Henk Muller (PDF)
 * [Hashing](https://www.smashwords.com/books/view/737188) - Prakash Hegade
+* [Learn C the hard way](https://users.elis.ugent.be/~jrvdnbro/learn_c_the_hard_way.pdf) - Zed A.Shaw (PDF)
```
Comment (eshellman) [inline free-programming-books.md:1062]: Not an authorized distribution. Please see CONTRIBUTING.md guidelines. please replace with Zed Shaw's authorized web site.

File: free-programming-books.md [hunk 3]
```diff
@@ -1068,8 +1069,10 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Programming in C](https://www.freetechbooks.com/programming-in-c-t1337.html) - Kishori Mundargi
 * [Structures and C](https://www.smashwords.com/books/view/644937) - Prakash Hegade
 * [The C book](http://publications.gbdirect.co.uk/c_book/) - Mike Banahan, Declan Brady and Mark Doran (PDF, HTML)
+* [The C Programming Language](https://inspirit.net.in/pdfviewer/web/viewer.html?file=..%2F..%2F.%2Fbooks%2Fc%20plus%20plus%2FThe%20C%20Programming%20Language%20-%20Brian%20Kernighan%20%26%20Dennis%20Ritchie.pdf) - Dennis M. Ritchie, Brian W. Kernighan (PDF)
 * [The Craft of Text Editing or A Cookbook for an Emacs](http://www.finseth.com/craft/) - Craig A. Finseth
 * [The Current C Programming Language Standard – ISO/IEC 9899:2018 (C17/C18), Draft](https://web.archive.org/web/20181230041359/http://www.open-std.org/jtc1/sc22/wg14/www/abq/c17_updated_proposed_fdis.pdf) - Open Standards Org - www.open-std.org (PDF)
+* [The Development of C Language](https://www.bell-labs.com/usr/dmr/www/chist.pdf) - Dennis M. Ritchie (PDF)
```
Comment (eshellman) [inline free-programming-books.md:1075]: Not a book.

**Review by eshellman — CHANGES_REQUESTED** [review]
Please read CONTRIBUTING.md guidelines. We want to encourage books that are free rather than unautorized postings that will be taken down, and possibly lead to legal trouble for us.

**Pull request closed without merging**